hello-security/list-users.php

```diff
@@ -0,0 +1,35 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <title>Login</title>
+</head>
+<body>
+<a href="index.php"> Take me home, country roads ... </a> <br>
+<?php
+session_start();
+if (!isset($_SESSION["email"])) {
+    die("Only authenticated users are allowed");
+}
+
+include "db-connection.php";
+$conn = openCon();
+
+$sql = "SELECT * from person where email = '" . $_SESSION["email"] . "';";
+$resultSet = $conn->query($sql) or die("Failed to query from DB!");
+$firstrow = $resultSet->fetch(PDO::FETCH_ASSOC) or die ("User not found.");
+
+if(strcmp($firstrow["role"], "admin") == 0) {
+	$sql = "SELECT * from person;";
+
+	$resultSet = $conn->prepare($sql);
+	$resultSet->execute();
```

Comment:
Great job to use prepare statement. In that way the code is not vulnerable to sql injection. It would be even better if you add die clause.